packages/fuels-programs/src/calls/script_call.rs

```diff
@@ -81,7 +92,8 @@ where
             script_binary,
             encoded_args,
             inputs: vec![],
-            outputs: vec![],
+            message_outputs: vec![],
+            variable_outputs: vec![],
```

Comment:
Is there some benefit from keeping message outputs and variable outputs separate in both `ContractCall` and `ScriptCall`? 

We couldn't find a reason why they couldn't just be one collection of `Outputs`. 

AFAIK the only restriction is to have contract outputs first because that is the way we've populated the contract input indexes. The rest can go in any order.

If that is true then these outputs don't need to be separated and we can simplify the code while we're here.

Reply:
That was my initial reaction too. I tried building with both of them merged but it gets really messy downstream when you are actually building the transaction, because the code to extract only one type of message is ugly. 
Also it was already the case with contracts so I just imitated.
But I can try again and we can see how it looks.

Reply:
It actually also breaks the `estimate_tx_dependency` functions for a reason that's not immediate to diagnose. I suggest we put this into a separate issue because it would be beyond the scope of this PR.

Reply:
Nvm it works fine, I thought we had to separate the two for generating outputs but you're rigth, as long as contract outputs are here, we don't need to differentiate them.